homeassistant/components/sensor/fido.py

```diff
@@ -0,0 +1,196 @@
+                      MESSAGES, 'mdi:message-alert'],
+    'text_int_limit': ['International text limit',
+                       MESSAGES, 'mdi:message-alart'],
+    'text_int_remaining': ['Internaltional remaining',
+                           MESSAGES, 'mdi:message-alert'],
+    'talk_used': ['Talk time',
+                  MINUTES, 'mdi:cellphone'],
+    'talk_limit': ['Talk time limit',
+                   MINUTES, 'mdi:cellphone'],
+    'talt_remaining': ['Talk time remaining',
+                       MINUTES, 'mdi:cellphone'],
+    'talk_other_used': ['Other Talk time',
+                        MINUTES, 'mdi:cellphone'],
+    'talk_other_limit': ['Other Talk time limit',
+                         MINUTES, 'mdi:cellphone'],
+    'talt_other_remaining': ['Other Talk time remaining',
+                             MINUTES, 'mdi:cellphone'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_VARIABLES):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+JANRAIN_CLIENT_ID = "bfkecrvys7sprse8kc4wtwugr2bj9hmp"
+HOST_JANRAIN = "https://rogers-fido.janraincapture.com"
+HOST_FIDO = "https://www.fido.ca/pages/api/selfserve"
+LOGIN_URL = "{}/widget/traditional_signin.jsonp".format(HOST_JANRAIN)
+TOKEN_URL = "{}/widget/get_result.jsonp".format(HOST_JANRAIN)
+ACCOUNT_URL = "{}/v3/login".format(HOST_FIDO)
+BALANCE_URL = "{}/v2/accountOverview".format(HOST_FIDO)
+FIDO_DOLLAR_URL = "{}/v1/wireless/rewards/basicinfo".format(HOST_FIDO)
+USAGE_URL = "{}/v1/postpaid/dashboard/usage".format(HOST_FIDO)
+
+
+DATA_MAP = {'data': ('data', 'D'),
+            'text': ('text', 'BL'),
+            'mms': ('text', 'M'),
+            'text_int': ('text', 'SI'),
+            'talk': ('talk', 'V'),
+            'talk_other': ('talk', 'VL')}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Fido sensor."""
+    # Create a data fetcher to support all of the configured sensors. Then make
+    # the first call to init the data.
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        fido_data = FidoData(username, password)
+        fido_data.update()
+    except requests.exceptions.HTTPError as error:
+        _LOGGER.error(error)
+        return False
+
+    name = config.get(CONF_NAME)
+
+    sensors = []
+    for variable in config[CONF_MONITORED_VARIABLES]:
+        sensors.append(FidoSensor(fido_data, variable, name))
+
+    add_devices(sensors)
+
+
+class FidoSensor(Entity):
+    """Implementation of a Fido sensor."""
+
+    def __init__(self, fido_data, sensor_type, name):
+        """Initialize the sensor."""
+        self.client_name = name
+        self.type = sensor_type
+        self.entity_id = "sensor.{}_{}".format(name, sensor_type)
```

Comment:
I just want to keep the same string between the sensor entity_id and the monitored_variables list